Get primary LakeTable info from catalog will thrown exception.

### Search before asking

- [x] I searched in the [issues](https://github.com/apache/fluss/issues) and found nothing similar.


### Fluss version

0.8.0 (latest release)

### Please describe the bug 🐞

Get primary LakeTable info from catalog will thrown exception: 
```java
org.apache.fluss.exception.FlussRuntimeException: Failed to update metadata

	at org.apache.fluss.client.metadata.MetadataUpdater.updateMetadata(MetadataUpdater.java:291)
	at org.apache.fluss.client.metadata.MetadataUpdater.updateTableOrPartitionMetadata(MetadataUpdater.java:226)
	at org.apache.fluss.client.FlussConnection.getTable(FlussConnection.java:104)
	at org.apache.fluss.flink.catalog.Flink21Catalog.getTable(Flink21Catalog.java:88)
	at org.apache.fluss.flink.catalog.FlinkCatalogTest.testCreateAlreadyExistsLakeTable(FlinkCatalogTest.java:328)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:147)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:86)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:92)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:86)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:217)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:213)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:138)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:68)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:129)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:129)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
	at java.base/java.util.concurrent.ForkJoinTask.doExec$$$capture(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
Caused by: org.apache.fluss.exception.UnknownServerException: org.apache.fluss.exception.FlussRuntimeException: Failed to get tables '[fluss.lake_table$lake]'.
	at org.apache.fluss.server.coordinator.MetadataManager.getTables(MetadataManager.java:602)
	at org.apache.fluss.server.metadata.ZkBasedMetadataProvider.getTablesMetadataFromZK(ZkBasedMetadataProvider.java:55)
	at org.apache.fluss.server.metadata.TabletServerMetadataProvider.getTablesMetadataFromZK(TabletServerMetadataProvider.java:86)
	at org.apache.fluss.server.RpcServiceBase.processMetadataRequest(RpcServiceBase.java:514)
	at org.apache.fluss.server.tablet.TabletService.metadata(TabletService.java:298)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.apache.fluss.rpc.netty.server.FlussRequestHandler.processRequest(FlussRequestHandler.java:63)
	at org.apache.fluss.rpc.netty.server.FlussRequestHandler.processRequest(FlussRequestHandler.java:34)
	at org.apache.fluss.rpc.netty.server.RequestProcessor.processRequest(RequestProcessor.java:98)
	at org.apache.fluss.rpc.netty.server.RequestProcessor.run(RequestProcessor.java:70)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: org.apache.fluss.exception.TableNotExistException: Table 'fluss.lake_table$lake' does not exist.

```

### Solution

_No response_

### Are you willing to submit a PR?

- [x] I'm willing to submit a PR!

--- a/fluss-flink/fluss-flink-2.1/src/main/java/org/apache/fluss/flink/catalog/Flink21Catalog.java
+++ b/fluss-flink/fluss-flink-2.1/src/main/java/org/apache/fluss/flink/catalog/Flink21Catalog.java
@@ -18,8 +18,10 @@
 
 package org.apache.fluss.flink.catalog;
 
+import org.apache.fluss.flink.lake.LakeFlinkCatalog;
 import org.apache.fluss.metadata.TableInfo;
 
+import org.apache.flink.annotation.VisibleForTesting;
 import org.apache.flink.table.api.Schema;
 import org.apache.flink.table.catalog.CatalogBaseTable;
 import org.apache.flink.table.catalog.CatalogTable;
@@ -44,11 +46,29 @@ public class Flink21Catalog extends FlinkCatalog {
         super(name, defaultDatabase, bootstrapServers, classLoader, securityConfigs);
     }
 
+    @VisibleForTesting
+    public Flink21Catalog(
+            String name,
+            String defaultDatabase,
+            String bootstrapServers,
+            ClassLoader classLoader,
+            Map<String, String> securityConfigs,
+            LakeFlinkCatalog lakeFlinkCatalog) {
+        super(
+                name,
+                defaultDatabase,
+                bootstrapServers,
+                classLoader,
+                securityConfigs,
+                lakeFlinkCatalog);
+    }
+
     @Override
     public CatalogBaseTable getTable(ObjectPath objectPath)
             throws TableNotExistException, CatalogException {
         CatalogBaseTable catalogBaseTable = super.getTable(objectPath);
-        if (!(catalogBaseTable instanceof CatalogTable)) {
+        if (!(catalogBaseTable instanceof CatalogTable)
+                || objectPath.getObjectName().contains(LAKE_TABLE_SPLITTER)) {
             return catalogBaseTable;
         }
 

--- a/fluss-flink/fluss-flink-2.1/src/test/java/org/apache/fluss/flink/catalog/FlinkCatalog21Test.java
+++ b/fluss-flink/fluss-flink-2.1/src/test/java/org/apache/fluss/flink/catalog/FlinkCatalog21Test.java
@@ -1,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.fluss.flink.catalog;
+
+import org.apache.fluss.flink.lake.LakeFlinkCatalog;
+
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.catalog.Column;
+import org.apache.flink.table.catalog.DefaultIndex;
+import org.apache.flink.table.catalog.ResolvedSchema;
+import org.apache.flink.table.catalog.UniqueConstraint;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+/** Test for {@link Flink21Catalog}. */
+public class FlinkCatalog21Test extends FlinkCatalogTest {
+
+    @Override
+    protected FlinkCatalog initCatalog(
+            String catalogName,
+            String databaseName,
+            String bootstrapServers,
+            LakeFlinkCatalog lakeFlinkCatalog) {
+        return new Flink21Catalog(
+                catalogName,
+                databaseName,
+                bootstrapServers,
+                Thread.currentThread().getContextClassLoader(),
+                Collections.emptyMap(),
+                lakeFlinkCatalog);
+    }
+
+    protected ResolvedSchema createSchema() {
+        return new ResolvedSchema(
+                Arrays.asList(
+                        Column.physical("first", DataTypes.STRING().notNull()),
+                        Column.physical("second", DataTypes.INT()),
+                        Column.physical("third", DataTypes.STRING().notNull())),
+                Collections.emptyList(),
+                UniqueConstraint.primaryKey("PK_first_third", Arrays.asList("first", "third")),
+                Collections.singletonList(
+                        DefaultIndex.newIndex(
+                                "INDEX_first_third", Arrays.asList("first", "third"))));
+    }
+}

--- a/fluss-flink/fluss-flink-common/src/main/java/org/apache/fluss/flink/catalog/FlinkCatalog.java
+++ b/fluss-flink/fluss-flink-common/src/main/java/org/apache/fluss/flink/catalog/FlinkCatalog.java
@@ -124,13 +124,30 @@ public class FlinkCatalog extends AbstractCatalog {
             String bootstrapServers,
             ClassLoader classLoader,
             Map<String, String> securityConfigs) {
+        this(
+                name,
+                defaultDatabase,
+                bootstrapServers,
+                classLoader,
+                securityConfigs,
+                new LakeFlinkCatalog(name, classLoader));
+    }
+
+    @VisibleForTesting
+    public FlinkCatalog(
+            String name,
+            String defaultDatabase,
+            String bootstrapServers,
+            ClassLoader classLoader,
+            Map<String, String> securityConfigs,
+            LakeFlinkCatalog lakeFlinkCatalog) {
         super(name, defaultDatabase);
         this.catalogName = name;
         this.defaultDatabase = defaultDatabase;
         this.bootstrapServers = bootstrapServers;
         this.classLoader = classLoader;
         this.securityConfigs = securityConfigs;
-        this.lakeFlinkCatalog = new LakeFlinkCatalog(catalogName, classLoader);
+        this.lakeFlinkCatalog = lakeFlinkCatalog;
     }
 
     @Override

--- a/fluss-flink/fluss-flink-common/src/test/java/org/apache/fluss/flink/catalog/FlinkCatalogTest.java
+++ b/fluss-flink/fluss-flink-common/src/test/java/org/apache/fluss/flink/catalog/FlinkCatalogTest.java
@@ -22,6 +22,7 @@ import org.apache.fluss.config.Configuration;
 import org.apache.fluss.exception.IllegalConfigurationException;
 import org.apache.fluss.exception.InvalidPartitionException;
 import org.apache.fluss.exception.InvalidTableException;
+import org.apache.fluss.flink.lake.LakeFlinkCatalog;
 import org.apache.fluss.flink.utils.FlinkConversionsTest;
 import org.apache.fluss.server.testutils.FlussClusterExtension;
 import org.apache.fluss.utils.ExceptionUtils;
@@ -36,6 +37,7 @@ import org.apache.flink.table.catalog.CatalogMaterializedTable;
 import org.apache.flink.table.catalog.CatalogPartitionSpec;
 import org.apache.flink.table.catalog.CatalogTable;
 import org.apache.flink.table.catalog.Column;
+import org.apache.flink.table.catalog.GenericInMemoryCatalog;
 import org.apache.flink.table.catalog.IntervalFreshness;
 import org.apache.flink.table.catalog.ObjectPath;
 import org.apache.flink.table.catalog.ResolvedCatalogMaterializedTable;
@@ -58,8 +60,7 @@ import org.apache.flink.table.expressions.ResolvedExpression;
 import org.apache.flink.table.expressions.utils.ResolvedExpressionMock;
 import org.apache.flink.table.factories.Factory;
 import org.apache.flink.table.factories.FactoryUtil;
-import org.junit.jupiter.api.AfterAll;
-import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
@@ -78,6 +79,7 @@ import static org.apache.fluss.config.ConfigOptions.TABLE_DATALAKE_FORMAT;
 import static org.apache.fluss.flink.FlinkConnectorOptions.BUCKET_KEY;
 import static org.apache.fluss.flink.FlinkConnectorOptions.BUCKET_NUMBER;
 import static org.apache.fluss.flink.FlinkConnectorOptions.SCAN_STARTUP_MODE;
+import static org.apache.fluss.flink.adapter.CatalogTableAdapter.toCatalogTable;
 import static org.apache.fluss.flink.utils.CatalogTableTestUtils.addOptions;
 import static org.apache.fluss.flink.utils.CatalogTableTestUtils.checkEqualsIgnoreSchema;
 import static org.apache.fluss.flink.utils.CatalogTableTestUtils.checkEqualsRespectSchema;
@@ -101,7 +103,8 @@ class FlinkCatalogTest {
     private static final FlinkConversionsTest.TestRefreshHandler REFRESH_HANDLER =
             new FlinkConversionsTest.TestRefreshHandler("jobID: xxx, clusterId: yyy");
 
-    static Catalog catalog;
+    private Catalog catalog;
+    private MockLakeFlinkCatalog mockLakeCatalog;
     private final ObjectPath tableInDefaultDb = new ObjectPath(DEFAULT_DB, "t1");
 
     private static Configuration initConfig() {
@@ -110,7 +113,7 @@ class FlinkCatalogTest {
         return configuration;
     }
 
-    private ResolvedSchema createSchema() {
+    protected ResolvedSchema createSchema() {
         return new ResolvedSchema(
                 Arrays.asList(
                         Column.physical("first", DataTypes.STRING().notNull()),
@@ -128,7 +131,7 @@ class FlinkCatalogTest {
     private CatalogTable newCatalogTable(
             ResolvedSchema resolvedSchema, Map<String, String> options) {
         CatalogTable origin =
-                CatalogTable.of(
+                toCatalogTable(
                         Schema.newBuilder().fromResolvedSchema(resolvedSchema).build(),
                         "test comment",
                         Collections.emptyList(),
@@ -158,29 +161,36 @@ class FlinkCatalogTest {
         return new ResolvedCatalogMaterializedTable(origin, resolvedSchema);
     }
 
-    @BeforeAll
-    static void beforeAll() {
-        // set fluss conf
-        Configuration flussConf = FLUSS_CLUSTER_EXTENSION.getClientConfig();
-        catalog =
-                new FlinkCatalog(
-                        CATALOG_NAME,
-                        DEFAULT_DB,
-                        String.join(",", flussConf.get(BOOTSTRAP_SERVERS)),
-                        Thread.currentThread().getContextClassLoader(),
-                        Collections.emptyMap());
-        catalog.open();
-    }
-
-    @AfterAll
-    static void afterAll() {
-        if (catalog != null) {
-            catalog.close();
-        }
+    protected FlinkCatalog initCatalog(
+            String catalogName,
+            String databaseName,
+            String bootstrapServers,
+            LakeFlinkCatalog lakeFlinkCatalog) {
+        return new FlinkCatalog(
+                catalogName,
+                databaseName,
+                bootstrapServers,
+                Thread.currentThread().getContextClassLoader(),
+                Collections.emptyMap(),
+                lakeFlinkCatalog);
     }
 
     @BeforeEach
     void beforeEach() throws Exception {
+        // set fluss conf
+        Configuration flussConf = FLUSS_CLUSTER_EXTENSION.getClientConfig();
+
+        mockLakeCatalog =
+                new MockLakeFlinkCatalog(
+                        CATALOG_NAME, Thread.currentThread().getContextClassLoader());
+        catalog =
+                initCatalog(
+                        CATALOG_NAME,
+                        DEFAULT_DB,
+                        String.join(",", flussConf.get(BOOTSTRAP_SERVERS)),
+                        mockLakeCatalog);
+        catalog.open();
+
         // First check if database exists, and drop it if it does
         if (catalog.databaseExists(DEFAULT_DB)) {
             catalog.dropDatabase(DEFAULT_DB, true, true);
@@ -195,6 +205,13 @@ class FlinkCatalogTest {
                     .isPresent()) {
                 throw e;
             }
+        }
+    }
+
+    @AfterEach
+    void afterEach() {
+        if (catalog != null) {
+            catalog.close();
         }
     }
 
@@ -270,7 +287,7 @@ class FlinkCatalogTest {
         ResolvedSchema resolvedSchema = this.createSchema();
         CatalogTable table2 =
                 new ResolvedCatalogTable(
-                        CatalogTable.of(
+                        toCatalogTable(
                                 Schema.newBuilder().fromResolvedSchema(resolvedSchema).build(),
                                 "test comment",
                                 Collections.singletonList("first"),
@@ -306,6 +323,11 @@ class FlinkCatalogTest {
         CatalogTable table = this.newCatalogTable(options);
         catalog.createTable(lakeTablePath, table, false);
         assertThat(catalog.tableExists(lakeTablePath)).isTrue();
+        // get the lake table from lake catalog.
+        mockLakeCatalog.registerLakeTable(lakeTablePath, table);
+        assertThat((CatalogTable) catalog.getTable(new ObjectPath(DEFAULT_DB, "lake_table$lake")))
+                .isEqualTo(table);
+
         // drop fluss table
         catalog.dropTable(lakeTablePath, false);
         assertThat(catalog.tableExists(lakeTablePath)).isFalse();
@@ -363,7 +385,7 @@ class FlinkCatalogTest {
                         UniqueConstraint.primaryKey(
                                 "PK_first", Collections.singletonList("first")));
         CatalogTable origin =
-                CatalogTable.of(
+                toCatalogTable(
                         Schema.newBuilder().fromResolvedSchema(resolvedSchema).build(),
                         "test comment",
                         Collections.emptyList(),
@@ -648,7 +670,7 @@ class FlinkCatalogTest {
         ResolvedSchema resolvedSchema = this.createSchema();
         CatalogTable table2 =
                 new ResolvedCatalogTable(
-                        CatalogTable.of(
+                        toCatalogTable(
                                 Schema.newBuilder().fromResolvedSchema(resolvedSchema).build(),
                                 "test comment",
                                 Collections.singletonList("first"),
@@ -758,7 +780,7 @@ class FlinkCatalogTest {
         ObjectPath partitionedPath = new ObjectPath(DEFAULT_DB, "partitioned_table1");
         CatalogTable partitionedTable =
                 new ResolvedCatalogTable(
-                        CatalogTable.of(
+                        toCatalogTable(
                                 Schema.newBuilder().fromResolvedSchema(resolvedSchema).build(),
                                 "test comment",
                                 Collections.singletonList("first"),
@@ -845,7 +867,7 @@ class FlinkCatalogTest {
         ResolvedSchema schema = createSchema();
         CatalogTable partTable =
                 new ResolvedCatalogTable(
-                        CatalogTable.of(
+                        toCatalogTable(
                                 Schema.newBuilder().fromResolvedSchema(schema).build(),
                                 "partitioned table for stats",
                                 Collections.singletonList("first"),
@@ -973,5 +995,24 @@ class FlinkCatalogTest {
         CatalogBaseTable tableCreated = catalog.getTable(tablePath);
         checkEqualsRespectSchema((CatalogTable) tableCreated, table);
         catalog.dropTable(tablePath, false);
+    }
+
+    private static class MockLakeFlinkCatalog extends LakeFlinkCatalog {
+        private final GenericInMemoryCatalog catalog;
+
+        public MockLakeFlinkCatalog(String catalogName, ClassLoader classLoader) {
+            super(catalogName, classLoader);
+            catalog = new GenericInMemoryCatalog(catalogName, DEFAULT_DB);
+        }
+
+        @Override
+        public Catalog getLakeCatalog(Configuration tableOptions) {
+            return catalog;
+        }
+
+        void registerLakeTable(ObjectPath tablePath, CatalogTable table)
+                throws TableAlreadyExistException, DatabaseNotExistException {
+            catalog.createTable(tablePath, table, false);
+        }
     }
 }

